### PR TITLE
Infra/picker expandable overlay

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
**Please review but don't merge !!!**

This refactor in Picker component purpose is to stop using old TextField expandable API (what controls the picker modal show/hide) and instead use our new ExpandableOverlay component that is dedicated solely for that

This PR has few goals
- Gradually disconnect Picker from old TextField in preparation for using the new TextField
- Using a better infrastructure for the expandable overlay that supports show either modal or dialog - which will help us in private Picker

Few notes
- This PR is not perfect, I left some code in comment in purpose
- I ran over all use cases in example screen and made sure they work and still supported 
- I add some future notes for things that we better deprecate but didn't really handle any of them in this PR
- I removed a failing test that is not relevant anymore and TBH didn't test much to begin with

## Changelog
Refactor Picker component internally using Incubator.ExpandableOverlay for handling the Picker modal

### CODE REVIEW

File: src/components/picker/index.js [hunk 1]
```diff
@@ -366,98 +393,134 @@ class Picker extends Component {
...
+            <TextField
+              ref={forwardedRef}
+              {...textInputProps}
+              containerStyle={[paddings, margins, positionStyle, containerStyle]}
+              {...this.getAccessibilityInfo()}
+              importantForAccessibility={'no-hide-descendants'}
+              value={label}
+              // expandable
+              // renderExpandable={this.renderExpandableModal}
+              // onToggleExpandableModal={this.toggleExpandableModal}
```
Comment (M-i-k-e-l) [inline src/components/picker/index.js:518]: Do we want to forbid these (`false`, `noop`) so the user cannot do it within `textInputProps`?

File: src/incubator/expandableOverlay/index.tsx [hunk 2]
```diff
@@ -70,10 +90,18 @@ const ExpandableOverlay = (props: ExpandableOverlayProps, ref: any) => {
     );
   };
 
+  const renderOverlay = () => {
+    if (renderCustomOverlay) {
+      return renderCustomOverlay({expandableVisible, openExpandable: showExpandable, closeExpandable: hideExpandable});
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/incubator/expandableOverlay/index.tsx:95]: I'll change expandableVisible to visible
But I wanted to have openExpandable and closeExpandable to match the imperative methods names

File: src/components/picker/index.js [hunk 3]
```diff
@@ -366,98 +393,134 @@ class Picker extends Component {
...
     const label = this.getLabelValueText();
     const {paddings, margins, positionStyle} = modifiers;
 
+    const modalProps = {
+      animationType: 'slide',
+      transparent: Constants.isIOS && this.props.enableModalBlur,
+      enableModalBlur: Constants.isIOS && this.props.enableModalBlur,
+      onRequestClose: this.props.topBarProps?.onCancel,
+      onShow,
+      ...this.props.pickerModalProps
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/picker/index.js:493]: I don't think I understand, can you write here a suggestion of how it'll look

File: src/components/picker/index.js [hunk 4]
```diff
@@ -356,6 +363,26 @@ class Picker extends Component {
     this.setState({searchValue: ''});
   };
 
+  renderCustomModal = ({expandableVisible, openExpandable, closeExpandable}) => {
+    const {renderCustomModal, children} = this.props;
+    const {value} = this.state;
+
+    // TODO: Deprecate renderCustomModal prop
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/picker/index.js:370]: I guess I really want to remove it (: 
Removed 👍

File: src/incubator/expandableOverlay/index.tsx [hunk 5]
```diff
@@ -39,18 +50,32 @@ interface ExpandableOverlayMethods {
...
+    modalProps,
+    dialogProps,
+    showTopBar,
+    topBarProps,
+    renderCustomOverlay,
+    ...others
+  } = props;
   const [expandableVisible, setExpandableVisible] = useState(false);
   const showExpandable = useCallback(() => setExpandableVisible(true), []);
   const hideExpandable = useCallback(() => setExpandableVisible(false), []);
```
Comment (Inbal-Tish) [inline src/incubator/expandableOverlay/index.tsx:66]: We need to invoke the Modal/Dialog's 'onDismiss' in 'hideExpandable' so the user can pass a callback. 
And maybe Modal's 'onShow' in 'showExpandable'

File: src/incubator/expandableOverlay/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/incubator/expandableOverlay/index.tsx]: So why not have only `openExpandable` and `closeExpandable`?

File: src/components/picker/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/picker/index.js:492]: I think the `spread` was confusing (and added `enableModalBlur`):
```
const {
      useNativePicker,
      renderPicker,
      customPickerProps,
      containerStyle,
      testID,
      onShow,
      renderCustomModal,
      forwardedRef,
      modifiers,
      enableModalBlur,
      topBarProps,
      pickerModalProps
    } = this.props
```

File: src/incubator/expandableOverlay/index.tsx [hunk 6]
```diff
@@ -92,7 +97,12 @@ const ExpandableOverlay = (props: ExpandableOverlayProps, ref: any) => {
 
   const renderOverlay = () => {
     if (renderCustomOverlay) {
-      return renderCustomOverlay({expandableVisible, openExpandable: showExpandable, closeExpandable: hideExpandable});
+      return renderCustomOverlay({
+        visible: expandableVisible,
```
Comment (M-i-k-e-l) [inline src/incubator/expandableOverlay/index.tsx:101]: Why not name it as `visible` internally as well?

File: src/components/picker/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/picker/index.js:492]: Ohh, you meant destruct the props at the top. 
Got ya, will do

File: src/incubator/expandableOverlay/index.tsx (hunk 5 above)
Comment (ethanshar) [reply src/incubator/expandableOverlay/index.tsx]: Fixed

File: src/incubator/expandableOverlay/index.tsx (hunk 6 above)
Comment (ethanshar) [reply src/incubator/expandableOverlay/index.tsx]: Changed

**Review by M-i-k-e-l — APPROVED** [review]

**Pull request merged**